fingerprint/models.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional
+
+from django.conf import settings
+from django.contrib.sessions.models import Session
+from django.contrib.auth.models import AbstractBaseUser
+from django.db import models
+from django.dispatch import receiver
+from django.db.models.signals import post_save
+from django.contrib.auth import get_user_model
+
+
+class UserSession(models.Model):
+    # by default, django stores session data in database; however, we cannot rely on it,
+    # since other session backends may be used, plus django doesn't store user-session
+    # mapping, so we need to have additional table for that
+
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, blank=True, null=True, related_name='sessions')
+    session_key = models.CharField(max_length=40)
+    created = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return self.session_key
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=['session_key', 'user'], name='unique_user_session'),
+        ]
+        indexes = [
+            models.Index(fields=['user', '-created']),
+        ]
+
+    def get_value_display(self) -> str:
+        return self.session_key[:8]
```

Comment:
we not use this for __str__ and make __repr__ more detailed?
https://docs.djangoproject.com/en/4.2/ref/models/instances/#django.db.models.Model.get_FOO_display
get_FOO_display has special meaning in django, so this is why I'm not sure about current approach

No strong feelings about this, still seems kinda strange to me as I would just use this implementation in `__str__`  .

Reply:
You got it right actually! I wanted `get_FOO_display` because
> This method returns the “human-readable” value of the field
and here we do exactly this - we return human-readable value for admin only which is shorter and more prone to collisions.
`__str__` still returns "fair" value which I think is good for all other purposes except admin.